### ISSUE TITLE
pyqt5-qt5 fails on Windows

### ISSUE THREAD

**Issue by alexeagle** [bazelbuild/rules_python#1121]
Here's a repro:
https://buildkite.com/bazel/rules-python-python/builds/4471#0186c7e3-d2c7-486b-8959-c9936b0a3c66

failure looks like

```
  File "C:\b\b2ufo7zr\external\rules_python\python\pip_install\tools\wheel_installer\wheel.py", line 105, in unzip
    installer.install(
  File "C:\b\b2ufo7zr\external\pypi__installer\installer\_core.py", line 96, in install
    for record_elements, stream, is_executable in source.get_contents():
  File "C:\b\b2ufo7zr\external\pypi__installer\installer\sources.py", line 158, in get_contents
    assert record is not None, "In {}, {} is not mentioned in RECORD".format(
AssertionError: In PyQt5_Qt5-5.15.2-py3-none-win_amd64.whl, PyQt5_Qt5-5.15.2.dist-info/RECORD is not mentioned in RECORD

**Comment by alexeagle** [comment]
note, with the matching version of pip, a `pip install` of this package@version does work correctly, so there's something about the way rules_python wheel_installer.wheel.py calls it, or the way we've laid this particular package out on disk after downloading it

**Comment by f0rmiga** [comment]
When I downloaded the wheel and extracted it manually to inspect the `RECORD` file, I noticed that it has the path not normalized:

```
PyQt5_Qt5-5.15.2.dist-info/LICENSE,sha256=IATuPvgoKoX329A136z2PPA9VpU3vAhlX_6xQOo2ccU,44738
PyQt5_Qt5-5.15.2.dist-info/METADATA,sha256=PkFIKgS5amRlPGMkYvOqcJxBKu4TFUe4WykQc_w-9rA,552
PyQt5_Qt5-5.15.2.dist-info/WHEEL,sha256=HvbHOnfIIFGyPVIOEbfOdnEmUOBRPrAzrHWhdC-8b8M,95
PyQt5_Qt5-5.15.2.dist-info\RECORD,,
```

I assumed that the problem would be in the processing of this file, which I found to be a third-party library (installer). Upon checking the version being used by rules_python (0.6.0), I went to see if the changelog to the latest version (0.7.0) had anything related to this, and it does! https://installer.pypa.io/en/stable/changelog/#v0-7-0-mar-17-2023

I traced it back to the fix in the installer wheel in https://github.com/pypa/installer/pull/152, which fixes the reported issue https://github.com/pypa/installer/issues/149, which happens to point out the same wheel here!

I'll open a PR bumping installer to 0.7.0.
